### ISSUE TITLE
Investigate "green" movies

### ISSUE THREAD

**Issue by wbwakeman** [AllenInstitute/segmentation-labeling-app#76]
Try to find repro case
Try to understand and implement fix

Potentially look into hardware acceleration

**Comment by kschelonka** [comment]
@njmei Would you be able to take this one?

**Comment by njmei** [comment]
So a bit of preliminary sleuthing uncovered:

We are using x264 (AVC) and:

[Firefox support for AVC is dependent upon the operating system's built-in or preinstalled codecs for AVC and its container in order to avoid patent concerns.](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Video_codecs#AVC_H.264)

We maybe should ask folks for system specs including OS

**Comment by isaak-willett** [comment]
I can't create a job to investigate this as I don't have permissions in the same AWS where the lambdas are deployed. I'll let someone else pick this up or someone can help me with permissions.

**Comment by kschelonka** [comment]
Hmm, your IAM role should have admin privileges. Which account are you
using, and what exactly is the error message?

**Comment by kschelonka** [comment]
I would also recommend deploying a webpage locally rather than using a
ground truth job. It will be easier to reproduce, control, and debug.

**Comment by isaak-willett** [comment]
This is the account id I'm using: 606907419058. The error message is 
`400 AccessDeniedException: User: arn:aws:iam::606907419058:user/isaak.willett is not authorized to perform: iam:PassRole on resource: arn:aws:iam::606907419058:role/service-role/AmazonSageMaker-ExecutionRole-20200429T115429 Request ID: de5b921b-6e3a-4e2b-b49b-a85e8aef87ef`
This was using the predefined role. I also tried to create my own role with the right privileges for the buckets and it also gave me a 400 error.
I ended up just working on a local host.

**Comment by kschelonka** [comment]
That's unexpected. And this is just trying to make a job in the console using the roles already created?

**Comment by isaak-willett** [comment]
Yep, nothing fancy.

**Comment by isaak-willett** [comment]
Jerome was able to reproduce this error on his machine in Firefox see attached videos. People using different codecs might have different abilities to decode and encode the videos.
# Things to Investigate:
- [ ] Regen some more videos using VP9 codec and have them played on a Mac and something with integrated graphics
- [ ] Have Jerome fully force close Firefox before replaying the video
### Quicktime
![with_quicktime.png](https://images.zenhubusercontent.com/5dde9d05537fdc0001b7a121/40d58ce6-4b54-4e75-b65c-cead17a475be)
### With Hardware Acceleration
![with_Hardware_accel_firefox.png](https://images.zenhubusercontent.com/5dde9d05537fdc0001b7a121/744790f6-19d3-49e5-8a14-b060adf79581)
### Without Hardware Acceleration
![without_Hardware_accel_firefox.png](https://images.zenhubusercontent.com/5dde9d05537fdc0001b7a121/21f46750-f5fe-4493-bb89-cba36d69fa2f)
[Uploading... without_Hardware_accel_firefox.png ]

**Comment by isaak-willett** [comment]
The videos are now encoded using VP9 but this has created problems on my windows machine where they are no longer supported.

**Comment by njmei** [comment]
@isaak-willett It sounds like the next step would be to investigate if there might be something wrong with how we are converting our images to videos via `imageio-ffmpeg` then...

Some argument to the `imageio-ffmpeg` `write_frames` function that stand out to me are: `      pix_fmt_in`, `macro_block_size`, `size`.

**Comment by isaak-willett** [comment]
I changed the output pixel format to yuv420p and it seems to be working, there is however vertical striping which I'm not sure if it is my monitor. I'm sending the videos to Jerome to get them run on a Mac.

**Comment by kschelonka** [comment]
If it isn't working for you, then I think we need to pursue other options.
We have to support Windows machines.

**Comment by isaak-willett** [comment]
Changed the output format to yuv420p, we had a slight bug from previous implementation where input shape to ffmpeg is (width, height) and we were supplying (row, col) didn't catch this because we weren't looking at non square videos. I fixed this and tested with Natalia and Jerome, they can both open in browsers but not in Quicktime. Jerome tested on VLC and Quicktime as well as browsers and it plays in VLC but not Quicktime.
